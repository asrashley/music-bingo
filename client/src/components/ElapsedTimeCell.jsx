import React from 'react';
import PropTypes from 'prop-types';
import { Cell } from 'rsuite-table';
import { ElapsedTime } from './ElapsedTime';

export const ElapsedTimeCell = ({ dataKey, rowData, className, ...props }) => {
    if (rowData === undefined) {
        return <Cell dataKey={dataKey} {...props} />;
    }

    let spanClassName = '';
    if (typeof (className) === 'function') {
        spanClassName = className(rowData);
    } else if (typeof (className) === 'string') {
        spanClassName = className;
    }

    return (
        <Cell dataKey={dataKey} {...props}>
            <span className={spanClassName}>
                <ElapsedTime date={rowData[dataKey]} />
            </span>
        </Cell>
    );
};

ElapsedTimeCell.propTypes = {
    dataKey: PropTypes.string.isRequired,
    rowData: PropTypes.object,
    className: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.string
    ]),
};