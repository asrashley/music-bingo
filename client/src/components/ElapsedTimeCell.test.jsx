import { renderWithProviders } from "../../tests";

import { ElapsedTimeCell } from './ElapsedTimeCell';

describe('ElapsedTimeCell component', () => {
    const data = {
        pk: 5,
        duration: 30016,
        album: "The 50s 60 Classic Fifties Hits",
        filename: "01 Rock Around The Clock.mp3",
        title: "Rock Around The Clock",
        artist: "Bill Haley & His Comets"
    };

    it('matches snapshot', () => {
        const { asFragment } = renderWithProviders(
            <ElapsedTimeCell dataKey="pk" rowData={data} className="class-name" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('className can be a function', () => {
        const classNameFn = () => 'link-func-class';
        const { getBySelector } = renderWithProviders(
            <ElapsedTimeCell rowData={data} dataKey="pk" className={classNameFn} />);
        getBySelector('.link-func-class');
    });

    it('allows rowData to be null', () => {
        renderWithProviders(<ElapsedTimeCell dataKey="pk" />);
    });
});