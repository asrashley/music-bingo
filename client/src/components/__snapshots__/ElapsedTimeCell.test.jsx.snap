// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ElapsedTimeCell component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rs-cell"
    role="gridcell"
    style="width: 0px; height: 46px; z-index: 0; left: 0px;"
  >
    <div
      class="rs-cell-content"
      style="width: 0px; height: 46px;"
    >
      <span
        class="class-name"
      >
        <span
          class="elapsed-time"
        >
          0.005 second ago
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
